src/downloader/Downloader.java

```diff
@@ -0,0 +1,67 @@
+package downloader;
+
+import java.io.File;
+
+/**
+ * This class is used to download the traces for the simulator
+ * @author Alex
+ *
+ */
+public class Downloader {
+	private static Downloader INSTANCE = new Downloader();
```

Comment:
Please use singleton pattern for Downloader when you do the initialisation ofr Downloader class.